cmd/kubeadm/app/phases/controlplane/manifests_test.go

```diff
@@ -135,6 +135,32 @@ func TestCreateStaticPodFilesAndWrappers(t *testing.T) {
 	}
 }
 
+func TestCreatePrivilegedContainerForOpenStack(t *testing.T) {
+	// Creates a Master Configuration with OpenStack cloud provider
+	cfg := &kubeadmapi.MasterConfiguration{
+		KubernetesVersion: "v1.8.0",
+		CloudProvider:     "openstack",
+	}
+
+	k8sVersion, _ := version.ParseSemantic(cfg.KubernetesVersion)
+	specs := GetStaticPodSpecs(cfg, k8sVersion)
+
+	var assertions = []struct {
+		staticPodName string
+	}{
+		{staticPodName: kubeadmconstants.KubeAPIServer},
+		{staticPodName: kubeadmconstants.KubeControllerManager},
+	}
+
+	for _, assertion := range assertions {
+		spec, _ := specs[assertion.staticPodName]
+		privileged := spec.Spec.Containers[0].SecurityContext.Privileged
+		if *privileged != true {
+			t.Errorf("GetStaticPodSpecs did not enable privileged containers in %s pod", assertion.staticPodName)
+		}
```

Comment:
what about negative test scenario: privileged should be only if cloud provider == "openstack", but if not, it should be non-privileged.

Reply:
ack